simulation/traffic_simulator/src/api/api.cpp

```diff
@@ -317,13 +326,30 @@ bool API::updateEntityStatusInSim()
     simulation_interface::toMsg(status.action_status().twist(), status_msg.action_status.twist);
     simulation_interface::toMsg(status.action_status().accel(), status_msg.action_status.accel);
     entity_manager_ptr_->setEntityStatus(status.name(), status_msg);
+    if (entity_manager_ptr_->isEgo(status.name()) && getCurrentTime() <= 0) {
+      ego_entity_simulation_->setInitialStatus(status_msg);
+    }
   }
   return res.result().success();
 }
 
 bool API::updateFrame()
 {
-  std::optional<traffic_simulator_msgs::msg::EntityStatus> ego_status_before_update = std::nullopt;
+  if (ego_entity_simulation_) {
+    ego_entity_simulation_->update(
+      clock_.getCurrentSimulationTime(), clock_.getStepTime(),
+      entity_manager_ptr_->isNpcLogicStarted());
+    if (not entity_manager_ptr_->isEgoSpawned()) {
+      THROW_SEMANTIC_ERROR("Malformed state: ego simulated but not registered in entity manager.");
```

Comment:
`SEMANTIC ERROR` is a category that is thrown when an error occurs that is not grammatically incorrect but semantically wrong. (For example, when directed to a lanelet ID that does not exist).
If this is an error that is supposed to be a bug in the simulator, use `THROW_SIMULATION_ERROR.`

Also, if the user seeing this exception means that the simulator is broken (which I think is pretty much the case with `THROW_SIMULATION_ERROR`), then for the sake of speeding up bug fix.
Please include a sentence in the error message such as
``This exception is basically not supposed to be sent. Contact the developer as there is some kind of bug.``

Reply:
I've changed it according to your feedback -> 9c4287eada66ce130d7e9bd627f824f189945968